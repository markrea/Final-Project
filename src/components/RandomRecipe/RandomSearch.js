import React, { useState } from "react";
import getRandomRecipe from "../../requests/getRandomRecipe";
import "../../styles/Search.css";
import "../../styles/RandomRecipe.css";

const RandomSearch = ({ setSearchResults }) => {
  // const [value, setValue] = useState("");
  const handleClick = async (event) => {
    event.preventDefault();
    setSearchResults(await getRandomRecipe());
  };

  return (
    <div className="RandomRecipe" data-testid="buttonDiv">
      <button className="randomButton" type="button" onClick={handleClick}>
        Go!
      </button>
    </div>
  );
};
export default RandomSearch;
