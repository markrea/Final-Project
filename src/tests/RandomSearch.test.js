import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom/extend-expect";
import RandomSearch from "../components/RandomRecipe/RandomRecipe";

const mockOnClick = jest.fn();

describe("RandomSearch", () => {
  it("component renders correctly with props", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <RandomSearch />
      </MemoryRouter>
    );
    expect(asFragment).toMatchSnapshot();
  });
  it("renders a div", () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <RandomSearch />
      </MemoryRouter>
    );
    expect(getByTestId("buttonDiv")).toBeInTheDocument();
  });
  it("renders a button with text content 'Go!'", () => {
    const { getByRole } = render(
      <MemoryRouter>
        <RandomSearch />
      </MemoryRouter>
    );
    expect(getByRole("button")).toBeInTheDocument();
    expect(getByRole("button")).toHaveClass("randomButton");
    expect(getByRole("button")).toHaveTextContent("Go!");
  });
});
