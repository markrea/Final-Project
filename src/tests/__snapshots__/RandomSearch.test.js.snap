// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomSearch component renders correctly with props 1`] = `[Function]`;
